<a href="https://colab.research.google.com/github/tsangrebecca/BloomTech/blob/main/Orientation/Practice_Exploratory_Data_Analysis_Learners.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Practice: Exploratory Data Analysis

Follow the instructions for the following tasks. If a task is completed correctly, the assert statement will show no error in your code. This notebook is for practice purposes only and does not need to be turned in. 

Feel free to make changes to this document. If you would like to **save** any changes, make sure to click "Copy to Drive" above. 

____

**Task 1**

Import the Pandas and Seaborn libraries

In [5]:
#your code here
import pandas as pd
import seaborn as sns

**Task 2**

- Use seaborn's `.load_dataset()` to read in the `"penguins"` dataset. 
- Assign this to an object called `penguin_df`

In [6]:
#your code here
penguin_df = sns.load_dataset('penguins')
penguin_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
339  Gentoo     Biscoe             NaN            NaN                NaN   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  
0         3750.0    Male  
1         3800.0  Female  
2         3250.0  Female  
3            NaN     NaN  
4         3450.0  Female  
..           ...     ...  
339          NaN     NaN  
340       4850.0  Female  
341       5750.0    Male  
342       5200.0  Female  
343       5400.0    Male  

[344 rows x 7 columns]

In [7]:
#test cells
assert penguin_df.shape == (344, 7), "Have you imported the dataset using the seaborn load_dataset function?"

**Task 3**

How many total missing values are present in the dataset? 

In [8]:
#your code here
penguin_df.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

<details>
    <summary style= "display:list-item">  <b> Click on the triangle to see the answer. </b> </summary>

    19
</details>

**Task 4**

Drop all rows with missing values from the data frame. [Click on this link if you need to review the documentation.](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.dropna.html)

In [10]:
#your code here
penguin_df = penguin_df.dropna()
penguin_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
4    Adelie  Torgersen            36.7           19.3              193.0   
5    Adelie  Torgersen            39.3           20.6              190.0   
..      ...        ...             ...            ...                ...   
338  Gentoo     Biscoe            47.2           13.7              214.0   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  
0         3750.0    Male  
1         3800.0  Female  
2         3250.0  Female  
4         3450.0  Female  
5         3650.0    Male  
..           ...     ...  
338       4925.0  Female  
340       4850.0  Female  
341       5750.0    Male  
342       5200.0  Female  
343       5400.0    Male  

[333 rows x 7 columns]

In [11]:
assert  penguin_df.shape == (333, 7), "Check that you have dropped all rows with null values and that you have reassigned the change back to the penguin_df object"

**Task 5**

How many columns are float datatypes? 

In [13]:
#your code here
penguin_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 333 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            333 non-null    object 
 1   island             333 non-null    object 
 2   bill_length_mm     333 non-null    float64
 3   bill_depth_mm      333 non-null    float64
 4   flipper_length_mm  333 non-null    float64
 5   body_mass_g        333 non-null    float64
 6   sex                333 non-null    object 
dtypes: float64(4), object(3)
memory usage: 20.8+ KB



<details>
    <summary style= "display:list-item">  <b> Click on the triangle to see the answer. </b> </summary>

    4
</details>

**Task 6** 

What is the average body mass for all penguins in this dataset? 

In [15]:
penguin_df['body_mass_g'].describe()

count     333.000000
mean     4207.057057
std       805.215802
min      2700.000000
25%      3550.000000
50%      4050.000000
75%      4775.000000
max      6300.000000
Name: body_mass_g, dtype: float64


<details>
    <summary style= "display:list-item">  <b> Click on the triangle to see the answer. </b> </summary>

    Approximately 4207
</details>